hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/CreatePipelineSubcommand.java

```diff
@@ -55,7 +55,14 @@
 
   @Override
   public void execute(ScmClient scmClient) throws IOException {
-    if (type == HddsProtos.ReplicationType.CHAINED) {
+    // Once we support creating EC containers/pipelines from the client, the
+    // client should check if SCM is able to fulfil the request, and
+    // understands an EcReplicationConfig. For that we also need to have SCM's
+    // version here from ScmInfo response.
+    // As I see there is no way to specify ECReplicationConfig properly here
+    // so failing the request if type is EC, seems to be safe.
+    if (type == HddsProtos.ReplicationType.CHAINED
+        || type == HddsProtos.ReplicationType.EC) {
```

Comment:
Is there a similar EC config for the create container CLI that would need to be blocked?

Reply:
The create container subcommand from CLI does not have any parameters to define either a pipeline or a replication config so far. (CreateSubCommand is the implementing class a child of ScmSubcommand).

Reply:
After offline discussion, container create CLI currently does not allow specifying the container type so it does not need to be blocked.